### PR TITLE
Macro

### PR DESCRIPTION
**Pull request by umbernhard**
Implements MACRO for batch comparison audits on the backend. See [here](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=1443314) for more. We'll need some significant frontend/middle-end changes too, for instance the ability to ingest store batch-level results and samples, and the ability to select this kind of audit on the frontend.

### CODE REVIEW

File: tests/test_macro.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        }
+
+
+    yield Sampler('MACRO', seed, risk_limit, contests, batches)
+
+
+
+def test_max_error(sampler):
+
+    # this is kind of a hacky way to do this but ¯\_(ツ)_/¯
```
Comment (eventualbuddha) [inline tests/test_macro.py]: What's hacky with this? What would I do to make it not hacky?

File: tests/test_macro.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    computed_p, result = sampler.audit.compute_risk(sampler.contests, sampler.margins, sampler.batch_results, sample)
+
+    U = sampler.audit.compute_U(sampler.contests, sampler.margins, sampler.batch_results)
+    expected_p =  0.247688222
+
+    delta = abs(expected_p - computed_p)
+
+    assert delta < 10**-4, 'Incorrect p-value: Got {}, expected {}'.format(computed_p, expected_p)
+
+    assert result, 'Audit did not terminat but should have'
```
Comment (eventualbuddha) [inline tests/test_macro.py]:
```suggestion
    assert result, 'Audit did not terminate but should have'
```

File: audits/macro.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
+import math
+from scipy import stats
+from audits.audit import RiskLimitingAudit
+
+class MACRO(RiskLimitingAudit):
+
+    def __init__(self, risk_limit):
+        super().__init__(risk_limit)
```
[the remark this answers is not in the document]
Comment (umbernhard) [reply audits/macro.py]: Ah, I didn't realize python init'd the super class automatically. Will fix.

File: tests/test_macro.py (hunk 1 above)
Comment (umbernhard) [reply tests/test_macro.py]: I think that comment was before I settled on a testing strategy. seems like there might have been a better way to set up the test data from Stark's paper, but this works. I can take the comment out if you think this is fine.

**Pull request merged**